### ISSUE TITLE
Adblocking stops working when last private window is closed

### ISSUE THREAD

**Issue by jpalus** [qutebrowser/qutebrowser#5810]
**Version info**:

qutebrowser v1.14.0
Git commit: 
Backend: QtWebEngine (Chromium 80.0.3987.163)
Qt: 5.15.1

**Does the bug happen if you start with `--temp-basedir`?**:
Yes.

**Description**
Doing _some_ sequence of actions adblocking stops working.

**How to reproduce**

Took me a while to figure out reproducer sequence, might as well affect different one:
1. Start qutebrowser with adblock list updated
2. Open private window :open -p
3. Go to page with ads (that's basically any web page now)
4. Confirm ads are not shown
5. Close private window
6. Open private window again :open -p
7. Go to the same web page
8. Ads are not blocked

**Comment by The-Compiler** [comment]
This happened somewhere between 786198fde5bae03b003768f84bf577f718fd784d (good) and 35d116dd43ce131f9ac685826f073c698c030ca2 (bad), so was introduced by #4147 (or my changes after that).

cc @lufte

**Comment by lufte** [comment]
Wow, great finding @jpalus. I can take a look this weekend.

**Comment by jpalus** [comment]
Looks like it affects downloads as well -- selecting downloadable link in second private window does nothing.

**Comment by user202729** [comment]
It affects content.javascript.enabled as well (enables JavaScript even if it's previously set to disabled), and the linked pull request did not fix it.

The reproduce steps are roughly similar to the one described in the issue.

**Comment by The-Compiler** [comment]
@user202729 Reproduced with `qutebrowser --temp-basedir -s content.javascript.enabled false about:blank ':open -p https://www.whatismybrowser.com/detect/is-javascript-enabled'`, closing the private window, and doing `:open -p https://www.whatismybrowser.com/detect/is-javascript-enabled'` again.

Also seems to be caused by a commit between 786198f (good) and 35d116d (bad) - and indeed neither #5813 nor https://github.com/lufte/qutebrowser/commit/140c728cc475a929a22e2783a535304a94474e09 fixes it...

I haven't really looked into it yet, and I have no idea why this could be happening, given that JS is a per-page rather than per-profile setting. @lufte do you have any idea what could be going on there?

**Comment by lufte** [comment]
Uh-oh.

```python
    global_settings.init_settings()

    _init_site_specific_quirks()
    _init_devtools_settings()
```
This is what's called after initializing the profiles in `webenginesettings.init`. After reviewing this issue I assumed we needed to call all 3 of these after creating a new private profile, however:
1. The custom user agents set in `_init_site_specific_quirks` are preserved. Is it because the user agent is read by qutebrowser directly from the config (by the interceptor module) instead of QWebEngine reading it from the profile?
2. The custom javascript.enabled values set in `_init_devtools_settings` are also preserved. Is it because settings set with a pattern are not affected by this issue? I tested your steps but setting `content.javascript.enabled` only for `https://www.whatismybrowser.com/*`, and it was preserved. I couldn't figure out this behavior.

Assuming that we only need `global_settings.init_settings()`, I think the following patch is enough:
```
diff --git a/qutebrowser/config/websettings.py b/qutebrowser/config/websettings.py
index cc307dd75..0f4616c4b 100644
--- a/qutebrowser/config/websettings.py
+++ b/qutebrowser/config/websettings.py
@@ -287,6 +287,7 @@ def clear_private_data() -> None:
     if objects.backend == usertypes.Backend.QtWebEngine:
         from qutebrowser.browser.webengine import webenginesettings
         webenginesettings.init_private_profile()
+        webenginesettings.global_settings.init_settings()
     elif objects.backend == usertypes.Backend.QtWebKit:
         from qutebrowser.browser.webkit import cookies
         assert cookies.ram_cookie_jar is not None
```

I'll push all my pending changes to the current PR so they are not scattered all over the place. If we decide we still want the first approach (the one with `if not is_first_time:`) I'll make the appropiate changes.

**Comment by coiby** [comment]
> Uh-oh.
> 
> ```python
>     global_settings.init_settings()
> 
>     _init_site_specific_quirks()
>     _init_devtools_settings()
> ```
> 
> This is what's called after initializing the profiles in `webenginesettings.init`. After reviewing this issue I assumed we needed to call all 3 of these after creating a new private profile, however:
> 
> 1. The custom user agents set in `_init_site_specific_quirks` are preserved. Is it because the user agent is read by qutebrowser directly from the config (by the interceptor module) instead of QWebEngine reading it from the profile?
> 2. The custom javascript.enabled values set in `_init_devtools_settings` are also preserved. Is it because settings set with a pattern are not affected by this issue? I tested your steps but setting `content.javascript.enabled` only for `https://www.whatismybrowser.com/*`, and it was preserved. I couldn't figure out this behavior.
> 
> Assuming that we only need `global_settings.init_settings()`, I think the following patch is enough:
> 
> ```
> diff --git a/qutebrowser/config/websettings.py b/qutebrowser/config/websettings.py
> index cc307dd75..0f4616c4b 100644
> --- a/qutebrowser/config/websettings.py
> +++ b/qutebrowser/config/websettings.py
> @@ -287,6 +287,7 @@ def clear_private_data() -> None:
>      if objects.backend == usertypes.Backend.QtWebEngine:
>          from qutebrowser.browser.webengine import webenginesettings
>          webenginesettings.init_private_profile()
> +        webenginesettings.global_settings.init_settings()
>      elif objects.backend == usertypes.Backend.QtWebKit:
>          from qutebrowser.browser.webkit import cookies
>          assert cookies.ram_cookie_jar is not None
> ```
> 
> I'll push all my pending changes to the current PR so they are not scattered all over the place. If we decide we still want the first approach (the one with `if not is_first_time:`) I'll make the appropriate changes.

This alone couldn't fix the issue! But #5813 could fix this issue.
